src/optimal_image/compress/image.rs

```diff
@@ -32,8 +37,18 @@ impl<'a> VipsImage<'a> {
         path: S,
     ) -> Result<VipsImage<'a>, Box<Error>> {
         let path = CString::new(path)?;
-        let c =
-            unsafe { vips::vips_image_new_from_file(path.as_ptr(), NULL_LIST) };
+        let c = unsafe {
+            let mut out: *mut *mut vips::VipsImage =
```

Comment:
most of the functions in vips that expect a pointer to a vips image do actually expect a pointer to a pointer to a vipsImage, i think it is because of having a context for temporary images an free all of them together, or something like that, but is a common pattern in C.